snapcraft/internal/repo.py

```diff
@@ -284,7 +285,26 @@ def _setup_apt_cache(rootdir, sources, project_options):
     return apt_cache, progress
 
 
-def _fix_symlinks(debdir):
+def fix_pkg_config(root, pkg_config_file, prefix_trim=None):
+    """Opens a pkg_config_file and prefixes the prefix with root."""
+    pattern_trim = None
+    if prefix_trim:
+        pattern_trim = re.compile(
+            '^prefix={}(?P<prefix>.*)'.format(prefix_trim))
+    pattern = re.compile('^prefix=(?P<prefix>.*)')
+    for line in fileinput.input(pkg_config_file, inplace=True):
```

Comment:
I personnally tend to use fileinput as a contextmanager (easier to read, cleaner on opening/close files) as it's a more pythonish way of doing files handling in python3: https://docs.python.org/3/library/fileinput.html